ui/src/overview.tsx

```diff
@@ -321,27 +399,92 @@ function DemographicCharts({ cohort }: DemographicChartsProps) {
 
   const api = useContext(EntityCountsApiContext);
 
-  const barColours = [
-    "#003f5c",
-    "#2f4b7c",
-    "#665191",
-    "#a05195",
-    "#d45087",
-    "#f95d6a",
-    "#ff7c43",
-    "#ffa600",
-  ];
+  const generatePropertyString = (
+    property: ChartConfigProperty,
+    entityCountStruct: tanagra.EntityCountStruct
+  ) => {
+    let propertyString = "";
+    // TODO(neelismail): Remove property key check once API supports age.
+    const entityCountPropertyValue =
+      entityCountStruct.definition?.[
+        property.key === "age" ? "year_of_birth" : property.key
+      ];
+
+    if (entityCountPropertyValue) {
+      let value =
+        entityCountPropertyValue.int64Val ??
+        entityCountPropertyValue.stringVal ??
+        entityCountPropertyValue.boolVal;
+
+      if (value) {
+        // TODO(neelismail): Remove age handling once the API supports them.
+        if (property.key === "age" && typeof value === "number") {
+          value = new Date().getFullYear() - value;
+        }
+
+        const definedValue = value;
```

Comment:
For some reason, I have to create this `definedValue` in order to prevent JavaScript from complaining that `value may be undefined` within the `if (property.buckets) { ... }` block. I thought this was especially strange because on line 419 I do a check to ensure `value` is defined. Do you have any suggestions about how this should be handled?